Tests/OpenFeatureTests/MultiProviderTests.swift

```diff
@@ -0,0 +1,317 @@
+import Combine
+import XCTest
+
+@testable import OpenFeature
+
+final class MultiProviderTests: XCTestCase {
```

Comment:
Would be great to add a test in DeveloperExperienceTests as well, to test the OpenFeatureAPI path as well (including observing the events from it)